packages/node/src/ethereum/api.ethereum.ts

```diff
@@ -137,6 +137,44 @@ export class EthereumApi implements ApiWrapper<EthereumBlockWrapper> {
     this.supportsFinalization = supportsFinalization;
     this.chainId = network.chainId;
     this.name = network.name;
+
+    if (this.client instanceof WebSocketProvider) {
+      return;
+    }
+
+    logger.debug(
+      `checking maximum allowed batch request size for ${this.endpoint}...`,
+    );
+
+    // Find and set the maximum batch size allowed by JsonRpcBatchProvider
+    let batchSize = 10; // default maximum value
+    let minBatchSize = 1;
+    let maxBatchSize = batchSize;
+    let found = false;
+
+    while (!found && minBatchSize <= maxBatchSize) {
+      try {
+        // Test the current batch size with a dummy request (getBlock)
+        const batchPromises = [];
+        for (let i = 0; i < batchSize; i++) {
+          batchPromises.push(this.client.getBlock('latest'));
```

Comment:
Ideally we don't waste requests doing this on startup each time and instead do it while making useful requests